.asciidoctor/lib/copy-to-clipboard-inline-macro/extension.rb

```diff
@@ -0,0 +1,73 @@
+require 'asciidoctor/extensions' unless RUBY_ENGINE == 'opal'
+require 'base64'
+
+include ::Asciidoctor
+
+# A block macro that embeds a copy to clipboard icon and copy the content identified by given id (target)
+#
+# Usage
+#
+#    [[run]]
+#    ....
+#    mvn clean test
+#    ....
+#
+#    You can copy previous script copyToClipboard:run[]
+#
+#    This is an important token pass:q[<span id="token">`1234`</span>]copyToClipboard:token[] for you.
+class ClipboardInlineMacro < Extensions::InlineMacroProcessor
+  use_dsl
+
+  named :copyToClipboard
+
+  def process parent, target, attrs
+
+    label = ""
+    if attrs.has_key? 'label'
+      label = attrs['label']
+    end
+
+    extdir = ::File.join(::File.dirname __FILE__)
+    icon_name = 'clippy.svg'
+
+    icon_location = %(#{extdir}/#{icon_name})
+    %(
+        <button class="cpybtn" width="13" data-clipboard-action="copy" data-clipboard-target="##{target}">#{label}
+          <img src="#{ClipboardInlineMacro::img64(icon_location)}" width="13"/>
+        </button>
+    )
+  end
+
+  def self.img64(path)
+    File.open(path, 'rb') do |img|
+      'data:image/svg+xml;base64,' + Base64.strict_encode64(img.read)
+    end
+  end
+
+end
+
+class ClipboardAssetsDocinfoProcessor < Extensions::DocinfoProcessor
+  use_dsl
+  at_location :footer
+
+  def process doc
+
+    clipboard = %(
+      var clipboard = new Clipboard('.cpybtn');
+      clipboard.on('success', function(e) {
+        console.log(e);
```

Comment:
Do we really have to log to console? Just wondering what's the usefulness of it

Reply:
I read in clipboard website. What I suspect is that it is important also to notice if something has been copied as well, so anyone can validate if the thing you are copying is the one it is on web page. For this reason I decided to left.

Reply:
You only see it when you have web tools opened in the browser, so normally you won't see it anyway.

Reply:
yes, but for example I see that might help in case any user opens an issue about copy to clipboard feature. We can always say, check console and tell me what's there or something like this. I see some small benefits on having it there.

Reply:
Ok, then maybe instead of having the same logging content let's differentiate if it was a `success` or a `failure`? e.g. use some prefixes so we know from where this log is coming, as well as `console.error` for the faulty path.

On top of that not all the versions of browsers are supporting logging - see http://caniuse.com/#feat=console-basic, but I'm not sure if we should care at this stage.